check.rb

```diff
@@ -67,7 +68,8 @@ def check_all
         client.authorizations.find { |a| a[:note] == GITHUB_AUTHORIZATION_NOTE } ||
         client.create_authorization(
           scopes: ['user:email'],
-          note: GITHUB_AUTHORIZATION_NOTE)
+          note: GITHUB_AUTHORIZATION_NOTE,
+          headers: {'X-Github-OTP': GITHUB_AUTHORIZATION_OTP})
```

Comment:
Thanks for taking a stab at it! Did you try it enabling 2-factor on your own GitHub account? I can't see how thsi can work, for me you need to send a value coming from the OTP generator in the `X-GitHub-OTP` header! That means having a third `puts "If you have 2-factor enabled, please type the 6-digits code, or just press <Enter> if you don't"` and then having a `gets.chomp` reading that. That way, if it's `""`, you don't send the `X-Github-OTP` header, if it has a value then you do send it!

Reply:
![image](https://user-images.githubusercontent.com/26072651/88821069-1387a380-d1ba-11ea-8b9b-b99393dc19ca.png)

Now just need to test if it's still works without 2FA 😅

Reply:
Works without too 🙌 
![image](https://user-images.githubusercontent.com/26072651/88827493-efc85b80-d1c1-11ea-89a7-c156c0b68510.png)